src/client/dfuse/pil4dfs/int_dfs.c

```diff
@@ -0,0 +1,3106 @@
+		}
+
+		fd_dup2ed_Dest = query_fd_forward_dest(fd_Directed);
+		if (fd_dup2ed_Dest >= FD_FILE_BASE)	{
+			if (cmd == F_SETFD)
+				return 0;
+			else if (cmd == F_GETFL)
+				return file_list[fd_dup2ed_Dest - FD_FILE_BASE].open_flag;
+		}
+
+		if (fd_Directed >= FD_DIR_BASE) {
+			fd_Directed -= FD_DIR_BASE;
+			OrgFunc = 0;
+		} else if (fd_Directed >= FD_FILE_BASE) {
+			fd_Directed -= FD_FILE_BASE;
+			OrgFunc = 0;
+		}
+
+		if ((cmd == F_DUPFD) || (cmd == F_DUPFD_CLOEXEC))	{
+			if (fd_save >= FD_DIR_BASE)	{
+				if (fd_save == DUMMY_FD_DIR)	{
+					printf("ERROR> Unexpected fd == DUMMY_FD_DIR in "
+						"fcntl(fd, F_DUPFD / F_DUPFD_CLOEXEC)\n");
+					return (-1);
+				}
+
+				Next_Dirfd = find_next_available_dirfd();
+				memcpy(&dir_list[Next_Dirfd], &dir_list[fd_Directed],
+				       sizeof(struct DIRSTATUS));
+				dup_next = dir_list[fd_Directed].fd_dup_next;
+				dir_list[fd_Directed].fd_dup_next = Next_Dirfd;
+				dir_list[Next_Dirfd].fd_dup_pre = fd_Directed;
+				dir_list[Next_Dirfd].fd_dup_next = dup_next;
+				if (dup_next >= 0)
+					dir_list[Next_Dirfd].fd_dup_pre = Next_Dirfd;
+				dir_list[Next_Dirfd].ref_count = 1;
+				return (Next_Dirfd + FD_DIR_BASE);
+			} else if (fd_save >= FD_FILE_BASE) {
+				Next_fd = find_next_available_fd();
+				memcpy(&file_list[Next_fd], &file_list[fd_Directed],
+				       sizeof(struct FILESTATUS));
+				dup_next = file_list[fd_Directed].fd_dup_next;
+				file_list[fd_Directed].fd_dup_next = Next_fd;
+				file_list[Next_fd].fd_dup_pre = fd_Directed;
+				file_list[Next_fd].fd_dup_next = dup_next;
+				if (dup_next >= 0)
+					file_list[Next_fd].fd_dup_pre = Next_fd;
+				file_list[Next_fd].ref_count = 1;
+				return (Next_fd + FD_FILE_BASE);
+			}
+		} else if ((cmd == F_GETFD) || (cmd == F_SETFD)) {
+			if (OrgFunc == 0)
+				return 0;
+		}
+/**		else if (cmd == F_GETFL)	{
+ *		}
+*/
+		return real_fcntl(fd, cmd, param);
+	case F_SETLK:
+	case F_SETLKW:
+	case F_GETLK:
+	case F_OFD_SETLK:
+	case F_OFD_SETLKW:
+	case F_OFD_GETLK:
+	case F_GETOWN_EX:
+	case F_SETOWN_EX:
+		va_start(arg, cmd);
+		param = va_arg(arg, int);
+		va_end(arg);
+
+		if (!inited)
+			return real_fcntl(fd, cmd, param);
+
+		return real_fcntl(fd, cmd, param);
+	default:
+		return real_fcntl(fd, cmd);
+	}
+
+	return real_fcntl(fd, cmd);
+}
+
+
+struct dfuse_user_reply {
+	uid_t uid;
+	gid_t gid;
+};
+
+int
+ioctl(int fd, unsigned long request, ...)
+{
+	va_list arg;
+	void *param;
+	struct dfuse_user_reply *reply;
+
+	va_start(arg, request);
+	param = va_arg(arg, void *);
+	va_end(arg);
+
+	if (real_ioctl == NULL)	{
+		real_ioctl = dlsym(RTLD_NEXT, "ioctl");
+		assert(real_ioctl != NULL);
+	}
+	if (!inited)
+		return real_ioctl(fd, request, param);
+
+	if (fd < FD_FILE_BASE)
+		return real_ioctl(fd, request, param);
+
+	if (request == 0xffffffff8008a3ca) {
+		/* DFUSE_IOCTL_DFUSE_USER */
+		reply = (struct dfuse_user_reply *)param;
+		reply->uid = getuid();
+		reply->gid = getgid();
+		return 0;
+	}
```

Comment:
There should be #define for the value but what is this code doing?  Is it useful to mock the ioctl but only fetch the uid and gid?

Reply:
I will remove this piece of code. It was added to test with the test case of libioil.so.